internal/services/appservice/windows_web_app_resource.go

```diff
@@ -911,7 +911,9 @@ func (r WindowsWebAppResource) Update() sdk.ResourceFunc {
 			if metadata.ResourceData.HasChanges("app_settings", "site_config") || metadata.ResourceData.HasChange("site_config.0.health_check_eviction_time_in_min") {
 				appSettingsUpdate := helpers.ExpandAppSettingsForUpdate(model.Properties.SiteConfig.AppSettings)
 				appSettingsProps := *appSettingsUpdate.Properties
-				appSettingsProps["WEBSITE_HEALTHCHECK_MAXPINGFAILURES"] = strconv.Itoa(int(state.SiteConfig[0].HealthCheckEvictionTime))
+				if metadata.ResourceData.HasChange("site_config.0.health_check_eviction_time_in_min") {
+					appSettingsProps["WEBSITE_HEALTHCHECK_MAXPINGFAILURES"] = strconv.Itoa(int(state.SiteConfig[0].HealthCheckEvictionTime))
+				}
```

Comment:
This still needs to be reverted to how it was previously in the code

Reply:
The "problem" is that without the if block, the app setting will be set to 0 when the health check property isn't configured in the terraform configuration.
Should I still remove the if block and create a different issue for that ?

Reply:
Given the complex and temperamental nature of App Service my preference would be to fix this in a separate PR with a dedicated test. You don't need to open a separate issue for it, just the PR is fine.